virtcontainers/sandbox.go

```diff
@@ -633,6 +633,80 @@ func (s *Sandbox) IOStream(containerID, processID string) (io.WriteCloser, io.Re
 	return c.ioStream(processID)
 }
 
+// PreAddDevices can let sandbox attach devices to hypervisor before container created
+// it will be mainly used by frakti to boost startup time by doing this in parallel
+func (s *Sandbox) PreAddDevices(contConfig ContainerConfig) ([]api.Device, error) {
+	var (
+		c       *Container
+		devices []api.Device
+		err     error
+	)
+
+	defer func() {
+		if err != nil {
```

Comment:
to reduce 1 indentation level and improve code readability, you should check `if err == nil` and return in all rolling back functions, what do you think?

Reply:
I see your point, nice suggestion. Will fix it